NUModules/NUModule.j

```diff
@@ -1235,11 +1276,37 @@ NUModuleTabViewModeIcon                = 2;
 */
 - (void)_loadNextPage
 {
-    // CS 01/11/2016: For now, we don't deal with pagination when using categories...
-    // Meaning that if we deal with pagination, we have only one context!
-    var contexts       = [self moduleCurrentActiveContexts],
-        fetcherKeyPath = [[contexts firstObject] fetcherKeyPath],
-        fetcher        = [_currentParent valueForKeyPath:fetcherKeyPath];
+    var fetcher;
+
+    if ([_categories count])
+    {
+        if (_maxPossiblePage != -1 && _latestPageLoaded >= _maxPossiblePage)
+        {
+            _latestCategoryLoadedIndex++;
```

Comment:
The idea is that when `_latestPageLoaded` reaches `_maxPossiblePage` we have fetched the entire category, so we reset both the values and go to the next one